src/coreclr/src/jit/optcse.cpp

```diff
@@ -1894,9 +1979,19 @@ class CSE_Heuristic
                     cost = dsc->csdTree->GetCostEx();
                 }
 
-                printf("CSE #%02u, {$%-3x, $%-3x} useCnt=%d: [def=%3u, use=%3u, cost=%3u%s]\n        :: ",
-                       dsc->csdIndex, dsc->csdHashKey, dsc->defExcSetPromise, dsc->csdUseCount, def, use, cost,
-                       dsc->csdLiveAcrossCall ? ", call" : "      ");
+                if ((dsc->csdHashKey & TARGET_SIGN_BIT) == 0)
+                {
+                    printf("CSE #%02u, {$%-3x, $%-3x} useCnt=%d: [def=%3u, use=%3u, cost=%3u%s]\n        :: ",
+                           dsc->csdIndex, dsc->csdHashKey, dsc->defExcSetPromise, dsc->csdUseCount, def, use, cost,
+                           dsc->csdLiveAcrossCall ? ", call" : "      ");
+                }
+                else
+                {
+                    size_t kVal = (dsc->csdHashKey & ~TARGET_SIGN_BIT) << CSE_CONST_SHARED_LOW_BITS;
```

Comment:
Nit: there are several uses of `(dsc->csdHashKey & ~TARGET_SIGN_BIT) << CSE_CONST_SHARED_LOW_BITS`, would be nice to have a helper with a header for this expression.

Reply:
Ok I can refactor this a well.